packages/common/src/internal-workflow/enums-helpers.ts

```diff
@@ -0,0 +1,294 @@
+import { Exact, RemovePrefix, UnionToIntersection } from '../type-helpers';
+
+/**
+ * Create encoding and decoding functions to convert between the numeric `enum` types produced by our
+ * Protobuf compiler and "const object of strings" enum values that we expose in our public APIs.
+ *
+ * ### Usage
+ *
+ * Newly introduced enums should follow the following pattern:
+ *
+ * ```ts
+ *     type ParentClosePolicy = (typeof ParentClosePolicy)[keyof typeof ParentClosePolicy];
```

Comment:
Yeah, kinda like we talked about in person, a macro really would make this dramatically easier.

I think probably some of this boilerplate could be reduced potentially, needing to pass the type, `typeof`, and `keyof` all separately, and possibly auto-assigning the enum numbers... but I don't think it's necessarily worth doing to save a few lines either.

Reply:
>Yeah, kinda like we talked about in person, a macro really would make this dramatically easier.

Definitely. Unfortunately, TypeScript is only about the type system. It can only be used to define type constraints, but can't define new code in any way. Hence this very 

> I think probably some of this boilerplate could be reduced potentially, needing to pass the type, typeof, and keyof all separately, and possibly auto-assigning the enum numbers... but I don't think it's necessarily worth doing to save a few lines either.

I thought similarly, and made several iterations already on that type definition, but for various reasons, that would each time end up in less strict validations.